src/CaptureServiceBase/include/CaptureServiceBase/StartStopCaptureRequestWaiter.h

```diff
@@ -17,6 +18,16 @@ class StartStopCaptureRequestWaiter {
   virtual ~StartStopCaptureRequestWaiter() = default;
   [[nodiscard]] virtual orbit_grpc_protos::CaptureOptions WaitForStartCaptureRequest() = 0;
   virtual void WaitForStopCaptureRequest() = 0;
+
+  void SetStopCaptureReason(CaptureServiceBase::StopCaptureReason stop_capture_reason) {
+    stop_capture_reason_ = std::move(stop_capture_reason);
+  }
+  [[nodiscard]] std::optional<CaptureServiceBase::StopCaptureReason> GetStopCaptureReason() const {
+    return stop_capture_reason_;
+  }
+
+ private:
+  std::optional<CaptureServiceBase::StopCaptureReason> stop_capture_reason_;
```

Comment:
Probably as expected, `StartStopCaptureRequestWaiter` is getting more complicated than originally intended. For example, it's no longer an interface. I'm missing the context quite a bit, in particular around `CloudCollectorStartStopCaptureRequestWaiter` and the structure of the cloud collector, so I'm not sure if there's clearly a better way really. Maybe @beckerhe has more thoughts.